src/chains/ethereum/ethereum/src/block-time.ts

```diff
@@ -0,0 +1,97 @@
+export interface BlockTime {
```

Comment:
I've extracted the logic for block timestamp into BlockTime, which allows us to more clearly describe the behaviour, and test more thoroughly.

By settings this as a dependency, it will allow us to decouple our tests from the system clock, for the likes of #3271 (which I'll pick up after this).

Reply:
I opened PR #3389 to address #3271 but have also fixed the issue in this PR, and was able to test the issue in this PR (at least on the underlying provider, not so easy at the higher level integration test).

I figured I would leave both PRs open, so that we can get a quick fix in with #3389, but am happy to close that PR if this one goes through quickly.